superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx

```diff
@@ -275,9 +275,10 @@ const FILTER_TYPE_NAME_MAPPING = {
 };
 
 // TODO: add column_types field to DatasourceMeta
+// We return true if column_types is undefined as a precaution against backend failing to return column_types
 const hasTemporalColumns = (
   dataset: DatasourceMeta & { column_types: GenericDataType[] },
-) => dataset?.column_types?.includes(GenericDataType.TEMPORAL);
+) => dataset?.column_types?.includes(GenericDataType.TEMPORAL) ?? true;
```

Comment:
Could we make it even more bulletproof by checking for length of `column_types`? Something like `!dataset?.column_types?.length || dataset?.column_types?.includes(GenericDataType.TEMPORAL)`? The reason is I suspect the Druid connector might return an empty array.

Reply:
Done 👍